src/app/token-option-field-component-factories/earn-by-survey-token-option-field-component-factory.ts

```diff
@@ -16,35 +16,81 @@ import {
     type EarnBySurveyTokenOption,
     type EarnBySurveyTokenOptionData
 } from 'app/token-options/earn-by-survey-token-option';
+import { QualtricsService } from 'app/services/qualtrics.service';
 
 @Injectable()
 export class EarnBySurveyTokenOptionFieldComponentFactory extends TokenOptionFieldComponentFactory<EarnBySurveyTokenOption> {
+    constructor(@Inject(QualtricsService) private qualtricsService: QualtricsService) {
+        super();
+    }
+
     public override create(environmentInjector: EnvironmentInjector): [
         (viewContainerRef: ViewContainerRef) => void,
         // eslint-disable-next-line @typescript-eslint/no-explicit-any
         FormField<EarnBySurveyTokenOption | TokenOptionGroup, EarnBySurveyTokenOptionData, any>
     ] {
+        const surveyFieldNameComp = createComponent(StringInputFieldComponent, {
+            environmentInjector: environmentInjector
+        });
+        surveyFieldNameComp.instance.label = 'Survey Field Name for Respondent’s Email (from SSO)';
+
+        const surveyDetailsCombinedComp = createFieldComponentWithLabel(
+            StringInputFieldComponent,
+            'Qualtrics Survey ID',
+            environmentInjector
+        )
+            .appendComp(surveyFieldNameComp)
+            .editField((field) => {
+                // Survey ID validator
+                field.fieldA.validator = async (value) => {
+                    field.fieldA.errorMessage = undefined;
```

Comment:
Please avoid using `field` directly in the validator. This causes the validator relies on its closure to get access to `field`. Instead, you can use the `value` argument supplied to the validator, which also includes `field.destValue` in it.